### ISSUE TITLE
Can't join any lobby when using LC API

### ISSUE THREAD

**Issue by fabriciomj** [steven4547466/LC-API#12]
Currently using r2modman and I can't join lobbies when playing modded with LC API installed. Simply disabling LC API or using `v2.2.0` allows me to join.

Learned about that in these reddit posts:
[post1](https://www.reddit.com/r/lethalcompany/comments/18r2irl/comment/kf2gmzj/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button)
[post2](https://www.reddit.com/r/lethalcompany/comments/18riqe3/comment/kf29q22/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button)

**Comment by Lordfirespeed** [comment]
do you mean you 'can't join *vanilla* lobbies'?

could you post your logs?

**Comment by fabriciomj** [comment]
I guess these are vanilla servers, tried joining 4 different ones.
https://pastebin.com/raw/RuvvURMq

**Comment by Lordfirespeed** [comment]
Yeah. The latest version of LCAPI includes a networking bundle that makes the modded client incompatible with the unmodded one.

@steven4547466 I think the best solution will be to add a config switch that disables all networking. Then throw an error if any mod tries to register any network messages.

**Comment by fabriciomj** [comment]
I mostly use client side mods and play pugs, so I would really like to be able to join vanilla lobbies. When I tried setting the modded server browser to true, I ran into the problem that I couldn't join the lobbies because I didn't have their server side mods and it is much harder to figure out what mods I need when joining randoms.

**Comment by steven4547466** [comment]
> @steven4547466 I think the best solution will be to add a config switch that disables all networking. Then throw an error if any mod tries to register any network messages.

The custom messages are actually fine, it's the player networking bundle that is the issue. Custom messages can be desynced.

But I can add a config option to disable any custom rpcs.

**Comment by Lordfirespeed** [comment]
Could we put LCApi's networking behaviour on a different game object then?? Would that enable vanilla interop?

**Comment by steven4547466** [comment]
> Could we put LCApi's networking behaviour on a different game object then?? Would that enable vanilla interop?

The main issue is I need to do it before the GO that has the `Player` behaviour is spawned, since mods like bigger lobby exists, I need to be able to dynamically instantiate and network spawn them into the scene. I will see if any object can fit this description

**Comment by Lordfirespeed** [comment]
How about adding a monobehaviour to the player prefab that will spawn the network object?

**Comment by steven4547466** [comment]
> How about adding a monobehaviour to the player prefab that will spawn the network object?

Players aren't prefabs, they exist in the scene just by default. I tried a bunch of stuff to reference them before the game is fully started, but I couldn't. And I need to add network behaviours before they are spawned by the host

**Comment by steven4547466** [comment]
However, there is now a config option for vanilla support out in v3.3.0, but I will keep this open so we can potentially find a better option

**Comment by ataknegn** [comment]
Hey. So me and my friends have tried to play the game with mods installed and one of them required LC_api to be downloaded. 4 of us downloaded it and 3 of us managed to join the server without any problems but the last one was unable to join no matter what. He reinstalled the game, mods multiple times and none worked so we tried deleting mods one by one and when we deleted LC_api he could join. I don't know what is the reason so a fix would be very nice.

Edit: The error in-game he gets is “An error occured” and the error in the console is “[Netcode] Network mismatch”

**Comment by steven4547466** [comment]
@ataknegn I will need the logs of at least 1 of the players that were able to join and the 1 that couldn't to diagnose that issue.
